plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java

```diff
@@ -89,21 +89,96 @@ public void setUp()
     }
 
     @Test
-    public void testBasicTypes()
+    public void testTrinoBoolean()
     {
         SqlDataTypeTest.create()
                 .addRoundTrip("boolean", "null", BOOLEAN, "CAST(NULL AS BOOLEAN)")
                 .addRoundTrip("boolean", "true", BOOLEAN)
                 .addRoundTrip("boolean", "false", BOOLEAN)
-                .addRoundTrip("bigint", "null", BIGINT, "CAST(NULL AS BIGINT)")
-                .addRoundTrip("bigint", "123456789012", BIGINT)
-                .addRoundTrip("integer", "null", INTEGER, "CAST(NULL AS INTEGER)")
-                .addRoundTrip("integer", "123456789", INTEGER)
-                .addRoundTrip("smallint", "null", SMALLINT, "CAST(NULL AS SMALLINT)")
-                .addRoundTrip("smallint", "32456", SMALLINT, "SMALLINT '32456'")
-                .addRoundTrip("tinyint", "null", TINYINT, "CAST(NULL AS TINYINT)")
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_boolean"));
+    }
+
+    @Test
+    public void testSqlServerBit()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("bit", "null", BOOLEAN, "CAST(NULL AS BOOLEAN)")
+                .addRoundTrip("bit", "1", BOOLEAN, "true")
+                .addRoundTrip("bit", "0", BOOLEAN, "false")
+                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_bit"));
+    }
+
+    @Test
+    public void testTinyint()
+    {
+        // TODO: Add min/max/min-1/max+1 tests (https://github.com/trinodb/trino/issues/11209)
+        SqlDataTypeTest.create()
+                .addRoundTrip("tinyint", "NULL", TINYINT, "CAST(NULL AS TINYINT)")
                 .addRoundTrip("tinyint", "5", TINYINT, "TINYINT '5'")
-                .execute(getQueryRunner(), trinoCreateAsSelect("test_basic_types"));
+                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_tinyint"))
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_tinyint"));
+    }
+
+    @Test
+    public void testSmallint()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("smallint", "NULL", SMALLINT, "CAST(NULL AS SMALLINT)")
+                .addRoundTrip("smallint", "-32768", SMALLINT, "SMALLINT '-32768'") // min value in SqlServer
+                .addRoundTrip("smallint", "32456", SMALLINT, "SMALLINT '32456'")
+                .addRoundTrip("smallint", "32767", SMALLINT, "SMALLINT '32767'") // max value in SqlServer
+                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_smallint"))
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_smallint"));
+    }
+
+    @Test
+    public void testUnsupportedSmallint()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_unsupported_smallint", "(data smallint)")) {
+            assertQueryFails(
+                    format("INSERT INTO %s VALUES (-32769)", table.getName()), // min - 1
+                    "Out of range for smallint: -32769");
```

Comment:
This error comes from Trino engine (not the connector nor SQL Server). Please execute the query on SQL Server to verify the actual limitation in the remote database.

Reply:
Updated.